Core/Src/serial_monitor.c

```diff
@@ -52,14 +60,17 @@ void vSerialMonitor(void *pv_params)
 		osMessageQueueNew(PRINTF_QUEUE_SIZE, sizeof(char *), NULL);
 
 	for (;;) {
-		/* Wait until new printf message comes into queue */
+		osThreadFlagsWait(NEW_MESSAGE_FLAG, osFlagsWaitAny,
+				  osWaitForever);
+
+		/* Get message to print */
 		status = osMessageQueueGet(printf_queue, &message, NULL,
 					   osWaitForever);
-		if (status != osOK) {
-			// TODO: Trigger fault ?
-		} else {
+		while (status == osOK) {
```

Comment:
why not inline osMessageQueueGet like you do elsewhere?

Reply:
This was definitely my first queue while loop i just forgot to go back good catch